src/js/helpers/ajaxWrapper.js

```diff
@@ -31,16 +31,23 @@ var qajaxWrapper = function (opts = {}) {
     body: null
   };
 
-  var parseResponse = function (xhr) {
+  var parseResponse = function (xhr, callback) {
     response.status = xhr.status;
-    try {
-      response.body = JSON.parse(xhr.responseText);
-    } catch (e) {
-      response.body = xhr.responseText;
-    }
+    xhr.json().then(function (body) {
+      response.body = body;
+    }, function () {
+      response.body = xhr.text().then(function (body) {
+        response.body = body
+      }, function () {
```

Comment:
https://github.com/mesosphere/marathon-ui/pull/476/files#r46976145 
... but I have to say the chained callback of ![image](https://cloud.githubusercontent.com/assets/1078545/11661306/8834aefc-9dd1-11e5-8bf1-40ab162784af.png) is also not the most readable thing ever ;)